### DISCUSSION TITLE
Compiling an agent.js compatible with frida-inject

### DISCUSSION THREAD

**Discussion by veryraregaming** [vfsfitvnm/frida-il2cpp-bridge#464] [Q&A] [answered]
Hello,

I've been successfully using frida-il2cpp-bridge on my PC, where it injects into the target android app, and my scripts function as intended. However, I'm encountering an issue with the `agent.js` file that's generated after compilation. It seems to be incompatible with `frida-inject`.

I suspect the problem lies in the size and content of the compiled JavaScript file, which includes emojis and numerous dependencies, making it excessively large. 

Could someone guide me on how to effectively utilize frida-il2cpp-bridge to develop scripts (using classes, methods, etc.) and compile a streamlined `agent.js`? The main challenge appears to be in the compilation process rather than the `index.ts` script itself.

Any advice on creating a clean, concise `agent.js` file that works smoothly with frida-inject would be greatly appreciated.

Thank you!

**Comment by Flechaa** [comment, marked as answer]
It's because ``frida-compile`` outputs a frida specific format that then frida tools compiles into actual JavaScript.

Your options are:
- Downgrade frida-compile to a version that doesn't have that feature
- Reimplement frida tools behavior then output it to a valid JavaScript file
- Use a different tool like [esbuild](https://esbuild.github.io/) with a plugin to implement frida shims if needed

**Comment by RevealedSoulEven** [comment]
Lol. From the beginning I'm using frida-inject library to inject using termux. I didn't used my pc till now as it needs to be repaired. From the beginning, I use this project on my android device using termux and frida-inject
